cpp/include/cudf/table/row_operators.cuh

```diff
@@ -107,127 +108,6 @@ __device__ weak_ordering relational_compare(Element lhs, Element rhs)
-__device__ bool equality_compare(Element lhs, Element rhs)
-{
-  if (isnan(lhs) and isnan(rhs)) { return true; }
-  return lhs == rhs;
-}
-
-/**
- * @brief A specialization for non-floating-point `Element` type to check if
- * `lhs` is equivalent to `rhs`.
- *
- * @param[in] lhs first element
- * @param[in] rhs second element
- * @return bool `true` if `lhs` == `rhs` else `false`.
- */
-template <typename Element, std::enable_if_t<not std::is_floating_point<Element>::value>* = nullptr>
-__device__ bool equality_compare(Element const lhs, Element const rhs)
-{
-  return lhs == rhs;
-}
-
-/**
- * @brief Performs an equality comparison between two elements in two columns.
- *
- * @tparam has_nulls Indicates the potential for null values in either column.
- **/
-template <bool has_nulls = true>
-class element_equality_comparator {
- public:
-  /**
-   * @brief Construct type-dispatched function object for comparing equality
-   * between two elements.
-   *
-   * @note `lhs` and `rhs` may be the same.
-   *
-   * @param lhs The column containing the first element
-   * @param rhs The column containing the second element (may be the same as lhs)
-   * @param nulls_are_equal Indicates if two null elements are treated as equivalent
-   **/
-  __host__ __device__ element_equality_comparator(column_device_view lhs,
-                                                  column_device_view rhs,
-                                                  bool nulls_are_equal = true)
-    : lhs{lhs}, rhs{rhs}, nulls_are_equal{nulls_are_equal}
-  {
-  }
-
-  /**
-   * @brief Compares the specified elements for equality.
-   *
-   * @param lhs_element_index The index of the first element
-   * @param rhs_element_index The index of the second element
-   *
-   */
-  template <typename Element,
-            std::enable_if_t<cudf::is_equality_comparable<Element, Element>()>* = nullptr>
-  __device__ bool operator()(size_type lhs_element_index, size_type rhs_element_index) const
-    noexcept
-  {
-    if (has_nulls) {
-      bool const lhs_is_null{lhs.nullable() and lhs.is_null(lhs_element_index)};
-      bool const rhs_is_null{rhs.nullable() and rhs.is_null(rhs_element_index)};
-      if (lhs_is_null and rhs_is_null) {
-        return nulls_are_equal;
-      } else if (lhs_is_null != rhs_is_null) {
-        return false;
-      }
-    }
-
-    return equality_compare(lhs.element<Element>(lhs_element_index),
-                            rhs.element<Element>(rhs_element_index));
-  }
-
-  template <typename Element,
-            std::enable_if_t<not cudf::is_equality_comparable<Element, Element>()>* = nullptr>
-  __device__ bool operator()(size_type lhs_element_index, size_type rhs_element_index)
-  {
-    release_assert(false && "Attempted to compare elements of uncomparable types.");
-    return false;
-  }
-
- private:
-  column_device_view lhs;
-  column_device_view rhs;
-  bool nulls_are_equal;
-};
-
-template <bool has_nulls = true>
-class row_equality_comparator {
```

Comment:
Why was this moved? `row_operators` is supposed to contain all of the operators that act on table rows.

Reply:
Good catch, @jrhemstad. I meant to move `element_equality_comparator` to a separate header for reuse. `row_equality_comparator` should have remained where it was. 

I have restored it to `row_operators.cuh`.